dsp/fourier/internal/fftpack/fftpack_test.go

```diff
@@ -14,116 +14,169 @@ import (
 	"reflect"
 	"testing"
 
+	"gonum.org/v1/gonum/cmplxs"
 	"gonum.org/v1/gonum/floats"
 	"gonum.org/v1/gonum/floats/scalar"
 )
 
 func TestRfft(t *testing.T) {
 	t.Parallel()
-	const tol = 1e-12
+	// Golden value cases where golden values have
+	// been obtained from the original Fortran code.
 	for _, test := range rfftTests {
-		// Compute the work and factor slices and compare to known values.
-		work := make([]float64, 2*test.n)
-		ifac := make([]int, 15)
-		Rffti(test.n, work, ifac)
-		var failed bool
-		if !floats.EqualApprox(work, test.wantiwork, 1e-6) {
-			failed = true
-			t.Errorf("unexpected work after call to rffti for n=%d", test.n)
-		}
-		if !reflect.DeepEqual(ifac, test.wantiifac) {
-			failed = true
-			t.Errorf("unexpected ifac after call to rffti for n=%d", test.n)
-		}
-		if failed {
-			continue
+		const tol = 1e-12
+		testRfft(t, test.n, tol, test.wantiwork, test.wantiifac)
+	}
+
+	// General cases based purely on FFT behaviour.
+	for n := 1; n < 500; n++ {
+		const tol = 1e-9
+		testRfft(t, n, tol, nil, nil)
+	}
+}
+
+func testRfft(t *testing.T, n int, tol float64, wantiwork []float64, wantiifac []int) {
+	// Compute the work and factor slices and compare to known values.
+	work := make([]float64, 2*n)
+	ifac := make([]int, 15)
+	Rffti(n, work, ifac)
+	var failed bool
+	if wantiwork != nil && !floats.EqualApprox(work, wantiwork, 1e-6) {
+		failed = true
+		t.Errorf("unexpected work after call to rffti for n=%d", n)
+	}
+	if wantiifac != nil && !reflect.DeepEqual(ifac, wantiifac) {
+		failed = true
+		t.Errorf("unexpected ifac after call to rffti for n=%d", n)
+	}
+	if failed {
+		return
+	}
+
+	// Construct a sequence with known a frequency spectrum and compare
+	// the computed spectrum.
+	modn := n % 2
+	fn := float64(n)
+	nm1 := n - 1
```

Comment:
This variable is used at only one place but it seems that this comes from the original Fortran code and I guess you plan to address these Fortranisms all at once.

Reply:
Same here.